lib/spack/spack/cmd/fetch.py

```diff
@@ -27,13 +28,24 @@ def setup_parser(subparser):
 
 
 def fetch(parser, args):
-    if not args.specs:
-        tty.die("fetch requires at least one package argument")
+    if args.specs:
+        specs = spack.cmd.parse_specs(args.specs, concretize=True)
```

Comment:
Other Spack commands will fail when you try to fetch a package that is not present in an environment. Should we do the same thing here?

Reply:
What would be the upside? I guess it would more clearly distinguish between in-environment and out-of-environment behaviour, but if I say `spack fetch <package>` I simply expect spack to fetch the thing and not annoy me with "technically this isn't in the environment"

Reply:
I agree with @healther on this one -- `fetch` doesn't affect environment state and shouldn't be affected by it.